native/NativeSampleApps/test/RestExplorerTest/src/com/salesforce/samples/restexplorer/ServerPickerActivityTest.java

```diff
@@ -113,39 +127,60 @@ public void testAddCustomInstance() throws Throwable {
         String label = "My Custom URL";
         String url = "https://valid.url.com";
         addCustomUrl(label, url);
-        clickView(com.salesforce.androidsdk.R.id.sf__apply_button);
-        openCustomEditDialog();
-        final CustomServerUrlEditor dialog = activity.getCustomServerUrlEditor();
-        Thread.sleep(3000);
-        final View rootView = dialog.getRootView();
-        final EditText txtLabel = rootView.findViewById(com.salesforce.androidsdk.R.id.sf__picker_custom_label);
-        final EditText txtUrl = rootView.findViewById(com.salesforce.androidsdk.R.id.sf__picker_custom_url);
-        Assert.assertTrue("Custom Label does not match Expected: " + label
-                + " Actual: " + txtLabel.getEditableText().toString(), label
-                .equalsIgnoreCase(txtLabel.getEditableText().toString()));
-        Assert.assertTrue("Custom URL does not match Expected: " + url + " Actual: "
-                + txtUrl.getEditableText().toString(), url
-                .equalsIgnoreCase(txtUrl.getEditableText().toString()));
+        checkServerAdded(label, url);
     }
 
     /**
-     * Test that "https" is required.
+     * Test that https is used if http is added.
      *
      * @throws Throwable
      */
     @Test
-    public void testAddInvalidUrl() throws Throwable {
-        String label = "My URL";
-        String url = "http://invalid.url.com";
-        addCustomUrl(label, url);
-        clickView(com.salesforce.androidsdk.R.id.sf__apply_button);
-        Assert.assertTrue("Custom URL dialog should still be open",
-                activity.getCustomServerUrlEditor().getDialog().isShowing());
-        url = "https://valid.url.com";
+    public void testAddHttpUrl() throws Throwable {
+        String label = "My http URL";
+        String httpUrl = "http://invalid.url.com";
+        String httpsUrl = "https://invalid.url.com";
+        addCustomUrl(label, httpUrl);
+        checkServerAdded(label, httpsUrl);
+    }
+
```

Comment:
Maybe one more test where you try to add a URL that has a protocol that is neither https nor http and you would check that nothing got added / that an error is shown?